File: backend/HrSystem.Domain/Entities/Employee.cs
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace HrSystem.Domain.Entities
{
    public class Employee
    {
        public int Id { get; set; }
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public string Position { get; set; } = string.Empty;
        public int DepartmentId { get; set; }
        public Department Department { get; set; } = null!;
        public decimal Salary { get; set; }
        public DateTime HireDate { get; set; }
    }
}

File: frontend/src/app/models/employee.ts
export interface Employee {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  position: string;
  departmentId: number;
  departmentName: string;
  salary: number;
  hireDate: string;
}

export interface CreateEmployee {
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  position: string;
  departmentId: number;
  salary: number;
  hireDate: string;
}
